core/src/test/java/org/apache/iceberg/TestSchemaParser.java

```diff
@@ -142,4 +143,31 @@ public void testPrimitiveTypeDefaultValues(Type.PrimitiveType type, Literal<?> d
     assertThat(serialized.findField("col_with_default").writeDefault())
         .isEqualTo(defaultValue.value());
   }
+
+  @Test
+  public void testVariantType() throws IOException {
```

Comment:
This is tested because `supportsVariantType` at the top is true. I would use the same pattern for spatial types, adding them to the list in `DataTest` and disabling the tests with a `supports` method. That way we can slowly add support and enable the tests (as we are doing for unknown, variant, and timestamp(9)).

Reply:
Removed `testVariantType` and updated `DataTest` to testing geospatial types. `supportsGeospatial` is added to `DataTest` and default to `false`. It is enabled in `TestSchemaParser`.